index.js

```diff
@@ -59,48 +48,62 @@ fs.readFile(file, 'utf8', (err, data) => {
-            searchBooks(books, title);
-
-            if (bookIndex >= 0) {
-                //if the book exists then we only need to push the highlight to the highlights array
-                books[bookIndex].highlights.push(highlight);
-            }
-            else {
-                //if book does not exist then create it
-                let book = createBook(title, highlight);
-                books.push(book);
-            }
-        }   
+          books[title] = [highlight];
+        }
     }
 
     // Loop through each book and write its contents to txt file
 
-    books.forEach((book) => {
-        txtFilePath = pathName + book.title + '.txt';
-        for (let i = 0; i < book.highlights.length; i++) {
-            fs.appendFile(txtFilePath, book.highlights[i] + "\n\n", (error) => {
-                if (error) {
-                    console.log(error);
-                }
-            });
+    Object.keys(books).forEach((title) => {
+      const highlights = books[title];
+      txtFilePath = pathName + title + '.txt';
+
+      for (let i = 0; i < highlights.length; i++) {
+        // conditional write function since we shouldn't append to the same files again and again when we run the script multiple times
+        function write(text) {
+          if (i === 0) {
+            fs.writeFile(txtFilePath, text + "\n\n", (err) => {
+              if (err) throw err;
+            })
+          } else {
+            fs.appendFile(txtFilePath, text + "\n\n", (err) => {
+              if (err) throw err;
+            })
+          }
         }
+        
+        write(highlights[i]);
+      }
     });
-    
-
 });
+
+// Using fs sync methods with utf8 encoding because I expect the file sizes won't be TOO big 
+// anything larger than (or even approaching) the memory allocation of the Node.js runtime will cause issues
+// that can be resolved by using readstreams and processing files chunk by chunk, but the logic is far trickier to write
+function verySimplified(inputFilePath = process.argv[2]) {
+  const booksDir = path.join(__dirname, "books");
+  // making our books dir if it doesn't exist
+  const booksDirExists = fs.existsSync(booksDir);
+  if (!booksDirExists) fs.mkdirSync(booksDir);
+
+  const imputFileText = fs.readFileSync(inputFilePath, "utf8");
+  const previouslySeenTitles = new Set();
+
+  // looping over entries from the input file
+  imputFileText.split(/={10}/).forEach(clipping => {
+    // parsing the title and highlight from each entry
+    const lines = clipping.split("\n").filter(line => line);
+    if (lines.length === 0) return;
+    const [title, _, highlight] = lines;
+
+    // conditionally writing/appending the highlight to output file
+    const seenBefore = previouslySeenTitles.has(title);
+    const bookFilePath = path.join(__dirname, "books", `${title}.txt`);
+    fs[seenBefore ? "appendFileSync" : "writeFileSync"](bookFilePath, `${seenBefore && "\n\n"}${highlight}`, "utf8");
```

Comment:
the ternary in the brackets on `fs` at the beginning of the line is just a fanciness. You don't see conditional method calls like this all that often. That said this felt like a reasonably appropriate place for such flare.